src/modules/endpoint/models/CategoriesContainer.js

```diff
@@ -0,0 +1,70 @@
+import normalizeUrl from 'normalize-url'
+
+/**
+ * Contains a Map [string -> CategoryModel] and some helper functions
+ */
+class CategoriesContainer {
+  /**
+   * Creates a Map [url -> category] from the categories provided,
+   * whose parent attributes are first changed from id to url
+   * @param categories CategoryModels as array
+   */
+  constructor (categories = []) {
+    this._categories = new Map(categories.map(category => ([category.url, category])))
+  }
+
+  /**
+   * @return {CategoryModel[]} categories The categories as array
+   */
+  toArray () {
+    return Array.from(this._categories.values())
+  }
+
+  /**
+   * Returns the category with the given url
+   * @param url The url
+   * @return {CategoryModel | undefined} The category
+   */
+  getCategoryByUrl (url) {
+    return this._categories.get(normalizeUrl(url))
+  }
+
+  /**
+   * Returns the category with the given id
+   * @param id The url
+   * @return {CategoryModel | undefined} The category
+   */
+  getCategoryById (id) {
+    return this.toArray().find(category => category.id === Number(id))
+  }
+
+  /**
+   * Returns all children of the given category
+   * @param category The category
+   * @return {CategoryModel[] | undefined} The children
+   */
+  getChildren (category) {
+    if (category) {
+      return this.toArray().filter(_category => _category.parentUrl === category.url)
```

Comment:
Don't we need to add `.sortBy(_category => _category.order)` here?

Reply:
Also why do you check if category is not null/undefined? We should do that before we call getChildren I think...

Reply:
I don't know, it's kind of strange but its sorted like it should, even without this sortBy (at least whenever I checked it, which are of course only samples). Should I add it anyway just to make sure?

Reply:
When there's no other place, where you sort by the order, then yes.

Reply:
done